docs/src/main/sphinx/connector/clickhouse.rst

```diff
@@ -34,6 +34,22 @@ appropriate for your setup:
     connection-user=exampleuser
     connection-password=examplepassword
 
+.. _clickhouse-tls:
+
+Connection security
+^^^^^^^^^^^^^^^^^^^
+
+If you have a globally-trusted certificate installed on your coordinator
+and all worker nodes, enable TLS between them by appending the ``ssl=true``
+parameter to the ``connection-url`` configuration property:
```

Comment:
TLS to data source doesn't care about what's in the Trino cluster. You can connect to SSL enabled data sources even without certificates on Trino or even if you use self-signed certs.

Reply:
Also `enable TLS between them` makes it sound like we are talking about TLS between worker <-> cordinator which isn't the case.

I'd instead prefer a short statement like "Consult the ClickHouse JDBC driver documentation regarding possible configurations" since our docs might go out of date and we aren't the source of truth for how to configure the client JDBC driver.

Reply:
Also note that how to configure SSL/TLS can vary across database versions or drivers being used. e.g. MySQL 8.x driver behaves differently compared to MySQL 5.x driver in how to configure the JDBC driver to use SSL.

Reply:
Great points, I think I have a way to reword these sections a bit that lets us discuss this topic on our docs while (reasonably) future-proofing them.

Reply:
Just removing the confusion between cluster level TLS vs data source TLS would be good enough for me + a stronger recommendation to look up driver docs to learn more.

Reply:
Reworded all these sections. I think the key information is just that you can configure TLS with parameters in `connection-url`, and to provide an example. Much more than that and we should offload to their respective JDBC docs.

The one exception in my edit is SingleStore/memsql because we still use the MariaDB JDBC driver rather than the official SingleStore driver (see https://github.com/trinodb/trino/issues/10669) and I don't know that we should call attention to that.